web/src/main/java/com/marklogic/hub/web/service/FlowJobService.java

```diff
@@ -103,13 +107,24 @@ private FlowJobs retrieveJobsByFlowName(String flowName) {
 
         final JsonNode[] lastJob = {null};
         final String[] lastTime = {null, null}; //store last start and end time
+        List<String> staleStateJobIds = new ArrayList<>();
         if (jsonNode.isArray()) {
             jsonNode.forEach(job -> {
                 JSONObject jobJson = new JSONObject(job.get("job"));
                 String jobId = jobJson.getString("jobId");
+                String jobStatus = jobJson.getString("jobStatus");
                 jobs.add(jobId);
                 String currStartTime = jobJson.getString("timeStarted", "");
                 String currEndTime = jobJson.getString("timeEnded", "");
+
+                if (cachedJobsByFlowName.size() == 0) {
+                    if (StringUtils.isNotEmpty(jobStatus) && jobStatus.startsWith(RUNNING_PREFIX)) {
```

Comment:
Is there any other scenario apart from during the initial login 'cachedJobsByFlowName' will be of size 0 ?

Reply:
I could not think of other scenarios...